pkg/reconciler/common/prune.go

```diff
@@ -249,7 +250,24 @@ func (pruner *Pruner) createJobContainers(nsConfig *pruneConfigPerNS, ns string)
 	return containers
 }
 
-func (pruner *Pruner) createCronJob(ctx context.Context, targetNs, schedule string, pruneContainers []corev1.Container, oR v1.OwnerReference) error {
+func (pruner *Pruner) checkAndCreate(ctx context.Context, cronName, targetNs, uniquePruneNs, schedule string, pruneContainers []corev1.Container, oR v1.OwnerReference, listOpt string) error {
+	cronList, err := pruner.listCronJobs(ctx, targetNs, listOpt)
+	if err != nil {
+		return err
+	}
+	if len(cronList.Items) == 1 {
+		if err := pruner.deleteCronJob(ctx, cronList.Items[0].Name, targetNs); err != nil {
+			return err
+		}
+		return createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR)
+	}
+	if err := createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR); err != nil {
+		return err
+	}
+	return nil
```

Comment:
```suggestion
	if len(cronList.Items) == 1 {
		if err := pruner.deleteCronJob(ctx, cronList.Items[0].Name, targetNs); err != nil {
			return err
		}
	}
	return createCronJob(ctx, pruner.kc, cronName, targetNs, uniquePruneNs, schedule, pruneContainers, oR)
```
as the eventual create actions is same from the if block and the normal flow, let us keep only one statement for create.

Reply:
not critical, :technologist: :+1: